pySDC/implementations/sweeper_classes/generic_implicit.py

```diff
@@ -65,6 +65,15 @@ def update_nodes(self):
         # get number of collocation nodes for easier access
         M = self.coll.num_nodes
 
+        # update the MIN-SR-FLEX preconditioner
```

Comment:
How specific is this to the generic implicit sweeper? Would it make sense to put this in a function `updatePreconditioner(self, k)` or whatever in the core Sweeper module that can be called in `update_nodes` of all sweepers? This irks me a bit because it is not really related to updating the nodes. We could consider putting this functionality in a convergence controller. There you wouldn't need the hack with the strings as well.
Having a preconditioner be a convergence controller is not great. If we can find a way of automatically adding this when the flex preconditioner is used in the sweeper parameters it might be cleaner than this, though. Again, sorry for not raising this point earlier. What do you think?

Reply:
This is indeed not generic at all, but I would leave it as it is for now : I will make it more generic when merging the explicit and implicit sweeper and regrouping the QDelta generation in a specific module out of the sweeper (because, there is no reason for the sweeper to be defined with QDelta matrices, one can also use a node-to-node formulation to define a sweeper, or an other approach that do not involve QDelta matrices).

This is just a quick solution which also makes sense for now : the FLEX preconditioning has been developed in an implicit only settings, so it makes sense to being able to use it with the implicit only sweeper.